contracts/core/IClient.sol

```diff
@@ -20,12 +20,33 @@ interface IClient {
         string calldata clientId
     ) external view returns (Height.Data memory, bool);
 
-    function checkHeaderAndUpdateState(
+    function verifyClientMessage(
```

Comment:
I understand that the interface functions(verifyClientMessage,updateState, checkForMisbehaviour, updateStateOnMisbehaviour) come from ibc-go, but I wonder if they should be added in ibc-solidity.

Since there is no abstract type like go's interface in solidity, client state needs to be represented as serialized bytes, except in the corresponding Light Client contract. Thus, this splitting of update_client function causes deserialization in each function. The costs are often high, so they are not negligible in solidity. So it is not necessary to apply such abstraction to the IClient.

Therefore, I think it would be better to simply rename `checkHeaderAndUpdateState` function as `updateClient` and each client implementation should handle misbehaviour in the function. Please let me know if I am missing something.

Reply:
hey yeah, I think your point is valid. I have renamed `checkHeaderAndUpdateState` to `verifyClientMessageAndUpdateState` and rolled back the corresponding changes.